File: src/homework/Nov26/miniProject/Page/AdminPage.java
package homework.Nov26.miniProject.Page;

import homework.Nov26.miniProject.State;
import homework.Nov26.miniProject.Abstract.Page;

public class AdminPage extends Page {
	
	public AdminPage() {}
	
	@Override
	public void show() {
		System.out.println("a: Add product\nb: Delete product\nc: Back");
		String input = State.getState().getScanner().next();
		switch (input) {
			case "a": {
				State.getState().getStack().push(this);
				new AddProductPage().show();
				break;
			}
			case "b": {
				State.getState().getStack().push(this);
				new DeleteProductPage().show();
				break;
			}
			case "c": {
				Page previousPage = (Page) State.getState().getStack().pop();
				previousPage.show();
				break;
			}
			default: {
				show();
				break;
			}
		}
	}
	
	
}

File: src/lab/miniProject_TS/src/page/adminPage.ts
import { Page } from "../interface"
import State from "../state"
import AddProductPage from "./addProductPage"
import DeleteProductPage from "./deleteProductPage"

class AdminPage implements Page {

    constructor() {}

    public show(): void {
        console.log('=========== Admin Mode ===========')
        const rl = State.getState().getRl()
        rl.question('a. Add product\nb. Delete product\nc. Back\n', (input: string) => {
            switch (input) {
                case 'a': {
                    State.getState().getStack().push(this);
                    new AddProductPage().show()
                    break;
                }
                case 'b': {
                    State.getState().getStack().push(this)
                    new DeleteProductPage().show()
                    break;
                }
                case 'c': {
                    State.getState().getStack().pop()?.show()
                    break
                }
                default: {
                    this.show()
                    break
                }
            }
        })
    }

}

export default AdminPage
